docs/designs/2730-visibility.md

```diff
@@ -0,0 +1,391 @@
+    verbosity level is set high enough
+  * Lead with indentation so output is visually obvious as something underneath
+    the top level group
+  * Review the messages that are printed by default so they are very clear and
+    easy for someone to understand from the context of where they are printed in
+    the overall output
+
+#### Cluster API Details
+
+Both the `management-cluster` and `cluster` creation operations utilize
+[cluster-api](https://cluster-api.sigs.k8s.io/) for declarative cluster creation.
+
+Since we have the `kubeconfig` for the bootstrap and management clusters, we can
+use this to gain more visibility into the cluster deployment process. We can get
+the [Cluster](https://cluster-api.sigs.k8s.io/developer/architecture/controllers/cluster.html)
+CRDs during deployment ot inspect the Status.Conditions to get the status of
+the new cluster creation. The JSON output from getting Cluster objects gives the
+following:
+
+```yaml
+"status": {
+      "conditions": [
+        {
+            "lastTransitionTime": "2022-01-31T20:29:39Z",
+            "status": "True",
+            "type": "Ready"
+        },
+        {
+            "lastTransitionTime": "2022-01-31T20:29:29Z",
+            "status": "True",
+            "type": "ControlPlaneInitialized"
+        },
+        {
+            "lastTransitionTime": "2022-01-31T20:29:39Z",
+            "status": "True",
+            "type": "ControlPlaneReady"
+        },
+        {
+            "lastTransitionTime": "2022-01-31T20:28:50Z",
+            "status": "True",
+            "type": "InfrastructureReady"
+        }
+      ],
+      "infrastructureReady": true,
+      "observedGeneration": 2,
+      "phase": "Provisioned"
+  }
+```
+
+By inspecting each of these statuses, we can report the progress of the cluster
+provisioning.
+
+Cluster-API also has `Bootstrap`, `Machine`, `MachineDeployment`, and other objects
+that can be retrieved to find more details. We can explore the attributes of these
+objects as well to see if there is any other status information to report to the
+user.
+
+Some of these Cluster-API objects will be too low level to show in the terminal
+output at default verbosity, but we could get their details to include in the
+bootstrap logging to aid in troubleshooting when deployments fail.
+
+#### Cluster API Controller Logs
+
+One common scenario we have found to work when troubleshooting deployment failures
+is to [look at the `capX-controller-manager` POD logs](https://tanzucommunityedition.io/docs/latest/tsg-bootstrap/#troubleshooting-manually).
+Looking for Error level logs (log lines that start with an `E`) will often indicate
+what is failing in reconcilation that is preventing the cluster from being
+properly created.
+
+Again, with having access to the `kubectl` for the cluster, we are able to
+automate this manual process, to some degree, by extracting these error logs
+and capturing them in the bootstrap log for easier troubleshooting.
+
```

Comment:
Can we include the machine health check controller as something we gather bootstrap logs from?

Reply:
Good call!